src/store/writer.rs

```diff
@@ -21,30 +23,64 @@ use crate::DocId;
 pub struct StoreWriter {
     compressor: Compressor,
     block_size: usize,
-    doc: DocId,
-    first_doc_in_block: DocId,
-    offset_index_writer: SkipIndexBuilder,
-    writer: CountingWriter<WritePtr>,
+    num_docs_in_current_block: DocId,
     intermediary_buffer: Vec<u8>,
     current_block: Vec<u8>,
+
+    // the channel to send data to the compressor thread.
+    compressor_sender: SyncSender<BlockCompressorMessage>,
+
+    // the handle to check for errors on the thread
+    compressor_thread_handle: JoinHandle<io::Result<()>>,
+}
+
+enum BlockCompressorMessage {
+    AddBlock(DocumentBlock),
+    Stack((StoreReader, DocumentBlock)),
```

Comment:
The semantics of Stack are very strange.
Can you comment on it? (and probably inverse the two args)

Reply:
Why cannot it be  `Stack(StoreReader)` and have the client emit two messages instead of one?

Reply:
split into two